core/src/main/java/com/oath/oak/OakAttachedReadBuffer.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2018 Oath Inc.
+ * Licensed under the terms of the Apache 2.0 license.
+ * Please see LICENSE file in the project root for terms.
+ */
+
+package com.oath.oak;
+
+import sun.nio.ch.DirectBuffer;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+class OakAttachedReadBuffer implements OakReadBuffer, OakUnsafeDirectBuffer {
```

Comment:
What OakBuffer implementation does it come to replace?

Reply:
I do not see it being used. Is it for future use?

Reply:
Yes. it is for future use.

Reply:
Regarding my first question, I understand now that this is instead of ByteBuffer we previously used in compare/transform/serialize.